marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/WriteDataTest.java

```diff
@@ -91,10 +94,24 @@ private void verifyFruitCount(int expectedCount, String message) {
         assertEquals(expectedCount, Integer.parseInt(count), message);
 
         if (expectedCount > 0) {
-            String expectedUri = "/testFruit/2/1.json";
-            JsonNode doc = getHubClient().getStagingClient().newJSONDocumentManager().read(expectedUri, new JacksonHandle()).get();
-            assertEquals("apple", doc.get("envelope").get("instance").get("fruitName").asText());
-            assertEquals("red", doc.get("envelope").get("instance").get("fruitColor").asText());
+        String uriQuery = "cts.uriMatch('/testFruit/**')";
+        EvalResultIterator uriQueryResult = getHubClient().getStagingClient().newServerEval().javascript(uriQuery).eval();
+        AtomicBoolean foundRedApple = new AtomicBoolean(false);
+        AtomicBoolean foundYellowBanana = new AtomicBoolean(false);
+        uriQueryResult.iterator().forEachRemaining(item -> {
+            final String expectUri = item.getString();
+            JsonNode doc = getHubClient().getStagingClient().newJSONDocumentManager().read(expectUri, new JacksonHandle()).get();
+            if((doc.get("envelope").get("instance").get("fruitName").asText().equals("apple")) &&
```

Comment:
This test is a little brittle in that if the structure of the document doesn't match, we'll get a null-pointer exception. 

Instead of doing a cts.uriMatch, I recommend doing a cts.uris query like this:

    cts.uris(null, [limit='1'], cts.andQuery([
      cts.directoryQuery("/testFruit/"),
      cts.jsonPropertyValueQuery("fruitName", someVariable)
    ]))

You can use newServerEval to execute that and then verify that you get 1 result back and that it starts with "/testFruit/". 

That results in a much simpler test and also much less chance of a null-pointer occurring.